### PR TITLE
Enable Github copilot and other goodies

### PR DESCRIPTION
**Pull request by EnriqueVidal**
- [x] Enable colorizer in every file
- [x] Replacing nvim-tree and web-dev-icons
- [x] Update lsp-sagas config
- [x] Add some QoL fucntions and keybindings for toggling color schemes
- [x] inlayhints and configure for tsserver and lua_ls

Inlayhints can be seen at the right of variables and LSP sagas can be seen displaying our location at the top editor

image

Copilot

https://user-images.githubusercontent.com/100161/228097422-1a3f1702-1de5-4399-8281-e4f8e3c80b37.mp4

### CODE REVIEW

File: .config/nvim/init.lua [hunk 1]
```diff
@@ -1,25 +1,28 @@
...
-  "plugins.nvim-tree", -- file explorer
-  "plugins.treesitter", -- treesitter
-  "plugins.telescope", -- fuzzy finder
-  "plugins.lsp.mason", -- automatic language server installer
-  "plugins.lsp.nvim-cmp", -- nvim completion
-  "plugins.lsp.lsp-config", -- laguage server protocol config
-  "plugins.lsp.null-ls", -- formatter and diagnostics
-  "plugins.gitsigns", -- git commands
+    -- plugins
+    "plugins.nvim-colorizer", -- Colorize hexs
```
Comment (aalvarado) [inline .config/nvim/init.lua:9]: what do you think about sorting these

File: .config/nvim/lua/plugins/lsp/lsp-config.lua [hunk 2]
```diff
@@ -15,98 +15,133 @@ vim.keymap.set("n", "<space>q", vim.diagnostic.setloclist, opts)
...
-  end, bufopts)
+    -- Mappings.
+    -- See `:help vim.lsp.*` for documentation on any of the below functions
+    local bufopts = { noremap = true, silent = true, buffer = bufnr }
+    vim.keymap.set("n", "gD", vim.lsp.buf.declaration, bufopts)
+    vim.keymap.set("n", "gd", vim.lsp.buf.definition, bufopts)
+    vim.keymap.set("n", "K", vim.lsp.buf.hover, bufopts)
+    vim.keymap.set("n", "gi", vim.lsp.buf.implementation, bufopts)
+    vim.keymap.set("n", "<C-k>", vim.lsp.buf.signature_help, bufopts)
+    vim.keymap.set("n", "<space>wa", vim.lsp.buf.add_workspace_folder, bufopts)
```
Comment (aalvarado) [inline .config/nvim/lua/plugins/lsp/lsp-config.lua:29]: should `<space>` be `<leader>`?

File: .config/nvim/lua/plugins/lsp/lsp-config.lua [hunk 3]
```diff
@@ -15,98 +15,133 @@ vim.keymap.set("n", "<space>q", vim.diagnostic.setloclist, opts)
...
+                library = vim.api.nvim_get_runtime_file("", true),
+            },
+            -- Enable ls hints
+            hint = {
+                enable = true,
+                setType = true,
+            },
+            -- Do not send telemetry data containing a randomized but unique identifier
+            telemetry = {
+                enable = false,
```
Comment (aalvarado) [inline .config/nvim/lua/plugins/lsp/lsp-config.lua:138]: sweet

File: .config/nvim/stylua.toml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+column_width = 80
+indent_type = "Spaces"
```
Comment (aalvarado) [inline .config/nvim/stylua.toml:2]: yeah, this is a must.

**Review by aalvarado — APPROVED** [review]
ship it

File: .config/nvim/lua/plugins/lsp/lsp-config.lua (hunk 2 above)
Comment (EnriqueVidal) [reply .config/nvim/lua/plugins/lsp/lsp-config.lua]: should keep in case leader changes.

**Pull request merged**